kubetest/e2e.go

```diff
@@ -173,12 +176,15 @@ func run(deploy deployer) error {
 	if *up {
 		// If we tried to bring the cluster up, make a courtesy
 		// attempt to bring it down so we're not leaving resources around.
-		//
-		// TODO: We should try calling deploy.Down exactly once. Though to
-		// stop the leaking resources for now, we want to be on the safe side
-		// and call it explictly in defer if the other one is not called.
 		if *down {
-			defer xmlWrap("Deferred TearDown", deploy.Down)
+			defer xmlWrap("Deferred TearDown", func() error {
+				if !downDone {
+					err := deploy.Down()
```

Comment:
since you're not setting `downDone` here, this could just be `return deploy.Down()`.